drivers/starburst/src/metabase/driver/implementation/query_processor.clj

```diff
@@ -221,3 +237,40 @@
   [_ _ expr]
   (let [report-zone (qp.timezone/report-timezone-id-if-supported :starburst)]
     (hsql/call :from_unixtime expr (hx/literal (or report-zone "UTC")))))
+
+(defmethod sql.qp/->honeysql [:starburst :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        disallowed-types (keep
+                          (fn [v]
+                            (when-let [db-type (keyword (hx/type-info->db-type (hx/type-info v)))]
+                              (let [base-type (sql-jdbc.sync/database-type->base-type driver db-type)]
+                                (when-not (some #(isa? base-type %) [:type/Date :type/DateTime])
+                                  (name db-type)))))
+                          [x y])]
+    (when (seq disallowed-types)
+      (throw (ex-info (tru "Only datetime, timestamp, or date types allowed. Found {0}"
+                           (pr-str disallowed-types))
+                      {:found disallowed-types
+                       :type  qp.error-type/invalid-query})))
+    (case unit
+      (:year :quarter :month :week :day)
+      (let [x-date (hsql/call :date (->AtTimeZone x (qp.timezone/results-timezone-id)))
+            y-date (hsql/call :date (->AtTimeZone y (qp.timezone/results-timezone-id)))]
+        (hsql/call :date_diff (hx/literal unit) x-date y-date))
+      (:hour :minute :second)
+      (hsql/call :date_diff (hx/literal unit) x y))))
+
+(defmethod sql.qp/->honeysql [:starburst :convert-timezone]
```

Comment:
looks good but same question as above, we need to either add tests or prove they already exist.

Reply:
https://github.com/metabase/metabase/blob/0975f5d9b2daf5737ad65fec8be4c3d4c65287dd/test/metabase/query_processor_test/date_time_zone_functions_test.clj#L496